File: cpp/028.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int s1 = haystack.size(), s2 = needle.size();
        for (int i = 0; i <= s1 - s2; ++i)
            if (haystack.substr(i, s2) == needle)
                return i;
        return -1;
    }
};

File: go/028.go
func strStr(haystack string, needle string) int {
    s1, s2 := len(haystack), len(needle)
    for i := 0; i <= s1-s2; i++ {
        if haystack[i:i+s2] == needle {
            return i
        }
    }
    return -1
}
